pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -33,7 +50,21 @@ func getClusterName(data *clusterNameData) string {
 
 	if !data.initDone {
 		data.clusterName = config.Datadog.GetString("cluster_name")
-		// TODO: autodiscover clustername through k8s providers' API
+		// autodiscover clustername through k8s providers' API
+		if data.clusterName == "" {
+			RegisterClusterNameProviders()
+			for _, getClusterNameFunc := range ProviderCatalog {
+				clusterName, err := getClusterNameFunc()
+				if err != nil {
+					// try the next cloud provider
+					continue
```

Comment:
As we are doing this once at agent start, we might want to `log.Debug` the error, in case autodetection were to fail because of missing instance permissions (thinking about ec2, but azure might also be impacted)

Log message should be non-alarming, as this is pretty optionnal. Here's what `GetHostname` logs for instance:

2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:128 in GetHostname) | Unable to get the hostname from the config file: hostname is empty
2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:129 in GetHostname) | Trying to determine a reliable host name automatically...
2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:138 in GetHostname) | GetHostname trying GCE metadata...
2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:149 in GetHostname) | Unable to get hostname from GCE:  unable to retrieve hostname from GCE: Get http://169.254.169.254/computeMetadata/v1/instance/hostname: dial tcp 169.254.169.254:80: connect: network is unreachable
2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:153 in GetHostname) | GetHostname trying FQDN/`hostname -f`...
2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:164 in GetHostname) | Unable to get FQDN from system:  <nil>
2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:181 in GetHostname) | GetHostname trying os...
2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:199 in GetHostname) | GetHostname trying EC2 metadata...
2019-02-27 13:59:32 UTC | DEBUG | (pkg/util/hostname.go:221 in GetHostname) | not retrieving hostname from AWS: the host is not an ECS instance, and other providers already retrieve non-default hostnames